src/js/components/ContainerSettingsComponent.jsx

```diff
@@ -0,0 +1,265 @@
+        volumes: [true]
+      }
+    };
+  },
+
+  handleAddRow: function (rowKey, position, event) {
+    event.target.blur();
+    event.preventDefault();
+    var rows = React.addons.update(
+      this.state.rows, {[rowKey]: {$push: [true]}}
+    );
+    this.setState({rows: rows});
+  },
+
+  handleRemoveRow: function (rowKey, position, event) {
+    event.target.blur();
+    event.preventDefault();
+    var rows = React.addons.update(
+      this.state.rows, {[rowKey]: {[position]: {$set: false}}}
+    );
+    if (rows[rowKey].filter((exists) => exists).length === 0) {
+      rows[rowKey].push(true);
+    }
+    this.setState({rows: rows});
+  },
+
+  getPortMappingRow: function (i = 0) {
+    return (
+      <div key={`pm-${i}`} className="row duplicable-row">
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].containerPort`}
+            label="Container Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].hostPort`}
+            label="Host Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].servicePort`}
+            label="Service Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].protocol`}
+            label="Protocol"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <select>
+              <option value="" disabled="disabled" selected="selected">Select</option>
+              <option value={ContainerConstants.PORTMAPPINGS.PROTOCOL.TCP}>
+                          {ContainerConstants.PORTMAPPINGS.PROTOCOL.TCP}
+              </option>
+              <option value={ContainerConstants.PORTMAPPINGS.PROTOCOL.UDP}>
+                          {ContainerConstants.PORTMAPPINGS.PROTOCOL.UDP}
+              </option>
+            </select>
+          </FormGroupComponent>
+          <DuplicableRowControls
+            handleAddRow={this.handleAddRow.bind(null, "portMappings", i)}
+            handleRemoveRow={this.handleRemoveRow.bind(null, "portMappings", i)} />
```

Comment:
To me it's not great to see `fn.bind(null, ...)`. Bind is meant to be used to bind a context as well as arguments. When we are only using it for args, it doesn't feel like it's being correctly utilized.
Also in terms of testing (if we were writing unit tests for this function) it's an odd way to write tests and set them up so that they work with the bound parameters.